### PR TITLE
feat(billing-metering): Billing and Metering to support plan crn in catalog offering

### PR DESCRIPTION
**Pull request by sreekarbvibm**

### CODE REVIEW

File: website/docs/r/is_instance.html.markdown [hunk 1]
```diff
@@ -417,6 +417,7 @@ resource "ibm_is_instance" "example" {
   profile = "cx2-2x4"
   catalog_offering {
     version_crn = data.ibm_is_images.example.images.0.catalog_offering.0.version.0.crn
+    plan_crn = "crn:v1:staging:public:globalcatalog-collection:global::1082e7d2-5e2f-0a11-a3bc-f88a8e1931fc:plan:sw.1082e7d2-5e2f-0a11-a3bc-f88a8e1931fc.7c7210da-bc40-418f-9a0a-af994040c13c-global"
```
Comment (ujjwal-ibm) [inline website/docs/r/is_instance.html.markdown:420]: missing documents for all resources and datasources

File: ibm/service/vpc/resource_ibm_is_volume.go [hunk 2]
```diff
@@ -614,6 +651,22 @@ func volGet(d *schema.ResourceData, meta interface{}, id string) error {
...
+		catalogMap[isVolumeCatalogOfferingVersionCrn] = versionCrn
+		if vol.CatalogOffering.Plan != nil {
+			catalogMap[isVolumeCatalogOfferingPlanCrn] = *vol.CatalogOffering.Plan.CRN
+			if vol.CatalogOffering.Plan.Deleted != nil {
+				deletedMap := resourceIbmIsVolumeCatalogOfferingVersionPlanReferenceDeletedToMap(*vol.CatalogOffering.Plan.Deleted)
+				catalogMap["deleted"] = []map[string]interface{}{deletedMap}
+			}
+		}
+		catalogList = append(catalogList, catalogMap)
+		d.Set(isVolumeCatalogOffering, catalogList)
```
Comment (ujjwal-ibm) [inline ibm/service/vpc/resource_ibm_is_volume.go:668]: set it empty outside nil check

File: ibm/service/vpc/resource_ibm_is_snapshot.go [hunk 3]
```diff
@@ -748,6 +785,23 @@ func snapshotGet(d *schema.ResourceData, meta interface{}, id string) error {
...
+		catalogList := make([]map[string]interface{}, 0)
+		catalogMap := map[string]interface{}{}
+		catalogMap[isSnapshotCatalogOfferingVersionCrn] = versionCrn
+		if snapshot.CatalogOffering.Plan != nil {
+			catalogMap[isSnapshotCatalogOfferingPlanCrn] = *snapshot.CatalogOffering.Plan.CRN
+			if snapshot.CatalogOffering.Plan.Deleted != nil {
+				deletedMap := resourceIbmIsSnapshotCatalogOfferingVersionPlanReferenceDeletedToMap(*snapshot.CatalogOffering.Plan.Deleted)
+				catalogMap["deleted"] = []map[string]interface{}{deletedMap}
+			}
+		}
```
Comment (ujjwal-ibm) [inline ibm/service/vpc/resource_ibm_is_snapshot.go:800]: set it outside nil as empty if comes empty

**Review by ujjwal-ibm — CHANGES_REQUESTED** [review]

**Comment by sreekarbvibm** [comment]
[config.tf.txt](https://github.com/ibm-vpc/terraform-provider-ibm/files/14442824/config.tf.txt)
terraform_apply_output
terraform_plan
[terraform.log](https://github.com/ibm-vpc/terraform-provider-ibm/files/14442826/terraform.log)

**Review by ujjwal-ibm — APPROVED** [review]

**Pull request closed without merging**